goblin.go

```diff
@@ -262,24 +266,30 @@ func runIt(g *G, it *It) {
 		//Set to nil as it shouldn't continue
 		g.shouldContinue = nil
 		g.timedOut = true
-		g.Fail("Test exceeded " + fmt.Sprintf("%s", g.timeout))
+		g.Fail("Test exceeded " + g.currentTimeout.String())
 	}
 	// Reset timeout value
-	g.timeout = *timeout
+	g.currentTimeout = g.defaultTimeout
 }
 
 type G struct {
 	t              *testing.T
 	parent         *Describe
 	currentIt      Itable
-	timeout        time.Duration
+	currentTimeout time.Duration
+	defaultTimeout time.Duration
 	reporter       Reporter
 	timedOut       bool
 	shouldContinue chan bool
 	mutex          sync.Mutex
 	timer          *time.Timer
 }
 
+func (g *G) SetDefaultTimeout(timeout time.Duration) {
```

Comment:
This function is never called / used?

Reply:
Correct, this is user-facing API.

Reply:
Ok, in that case this needs to be tested. However don't add anything until we decide if it makes sense to add this functionality or not. I still don't quite understand what this is improving / fixing.